src/Data/Lens/Types.purs

```diff
@@ -102,33 +111,46 @@ type Optic p s t a b = p a b -> p s t
 type Optic' :: (Type -> Type -> Type) -> Type -> Type -> Type
 type Optic' p s a = Optic p s s a a
 
+-- | An isomorphism defined in terms of `Exchange` to be safe from
+-- | impredicativity.
```

Comment:
Perhaps "to avoid issues with impredicativity" rather than "to be safe from"?

Reply:
Works for me.